src/app/api/challenge/submit/route.ts

```diff
@@ -1,66 +1,74 @@
-import { NextResponse } from 'next/server';
-import { createRedisClient, getChallengeFromCookie } from '@/app/api/utils';
+import { NextResponse } from "next/server";
+import { createRedisClient, getChallengeFromCookie } from "@/app/api/utils";
 
 export async function POST(request: Request) {
-    const redis = createRedisClient();
+  const redis = createRedisClient();
 
-    try {
-        const challengeStatus = await getChallengeFromCookie(redis);
+  try {
+    const challengeStatus = await getChallengeFromCookie(redis);
 
-        if (!challengeStatus) {
-            return NextResponse.json(
-                { success: false, message: 'Challenge not found' },
-                { status: 404 }
-            );
-        }
-
-        if (!challengeStatus.isStarted) {
-            return NextResponse.json(
-                { success: false, message: 'Challenge has not been started' },
-                { status: 400 }
-            );
-        }
+    if (!challengeStatus) {
+      return NextResponse.json(
+        { success: false, message: "Challenge not found" },
+        { status: 404 },
+      );
+    }
 
-        // Get the submitted answer from the request body
-        const { submission } = await request.json();
+    if (!challengeStatus.isStarted) {
+      return NextResponse.json(
+        { success: false, message: "Challenge has not been started" },
+        { status: 400 },
+      );
+    }
 
-        if (!submission) {
-            return NextResponse.json(
-                { success: false, message: 'No submission provided' },
-                { status: 400 }
-            );
-        }
+    const { submission } = await request.json();
 
-        if (!challengeStatus.startTime) {
-            return NextResponse.json(
-                { success: false, message: 'Challenge has not been started' },
-                { status: 400 }
-            );
-        }
+    if (!submission) {
+      return NextResponse.json(
+        { success: false, message: "No submission provided" },
+        { status: 400 },
+      );
+    }
 
-        // Calculate time taken
-        const endTime = new Date().getTime();
-        const timeTaken = endTime - challengeStatus.startTime;
+    const currentTime = new Date().getTime();
+    const isOvertime = currentTime > challengeStatus.endTime!;
 
-        // Update challenge status
-        challengeStatus.endTime = endTime;
-        challengeStatus.submission = submission;
+    if (isOvertime && challengeStatus.submission) {
+      return NextResponse.json(
+        { success: false, message: "Overtime submission not allowed" },
+        { status: 400 },
+      );
+    }
 
-        // Save updated challenge status
-        await redis.set(`challenge:${challengeStatus.emailAddress}`, JSON.stringify(challengeStatus));
+    // Update challenge status
+    challengeStatus.submission = submission;
```

Comment:
Can we change this to being a list of submissions that include timestamps for each? We can still show just the latest in the frontend

Reply:
Yeah, made the change